web/app/view/MapController.js

```diff
@@ -166,6 +182,43 @@ Ext.define('Traccar.view.MapController', {
                 if (marker === this.selectedMarker && this.followSelected()) {
                     this.getView().getMapView().setCenter(marker.getGeometry().getCoordinates());
                 }
+
+                if (deviceId in this.liveRoutes) {
+                    lastLiveCoordinates = this.liveRoutes[deviceId][this.liveRoutes[deviceId].length - 1].getGeometry().getCoordinates();
+                    if (lastLiveCoordinates[0] === geometry.getCoordinates()[0] &&
```

Comment:
I think we should get rid of "continue". Also, it would be nice to split this huge method into separate smaller methods for more readability.